wordpress-mobile/WordPress-iOS
# Implements the new relative date logic for reader card improvements

Subtask for #14185

The logic used (as defined by the `figma - card anatomy` section)

- If date is within the past hour display: `{Val}m`
	- `m` is an abbreviation for minutes 
	-  **Example:** 1m, 30m, 59m
- If date is  within the past 24 hours display: `{Val}h`
	- `h` is abbreviation for hours
	- **Example:** 1h, 5h, 20h
- If date is older than 24 hours but earlier than 7 days: `{Val}d`
	- `d` is an abbr for days
	- **Example:** 1d, 2d, 6d
- If date is older than 7 days but within the current year: `MMM DD`
	- **Example:** Feb 11, Jan 2
- If date is not from the current year `MMM DD, YYYY` or the users chosen `medium` style date format
	- **Example:** Feb 11 2019, Jan 2, 2010

### Screenshots
| Within an hour | Within 24 hours | With 7 days | Older than 7 days | Last Year |
|---|---|---|---|---|
| <img width="373" alt="Screen Shot 2020-05-25 at 9 15 06 PM" src="https://user-images.githubusercontent.com/793774/82859904-e45c6980-9ecc-11ea-8101-f5986791667b.png">|  <img width="375" alt="Screen Shot 2020-05-25 at 9 11 18 PM" src="https://user-images.githubusercontent.com/793774/82859777-7b74f180-9ecc-11ea-8e20-dc2a3448b9d7.png"> | <img width="372" alt="Screen Shot 2020-05-25 at 9 11 59 PM" src="https://user-images.githubusercontent.com/793774/82859788-816ad280-9ecc-11ea-8174-dad76a8490dd.png"> | <img width="374" alt="Screen Shot 2020-05-25 at 9 12 10 PM" src="https://user-images.githubusercontent.com/793774/82859795-862f8680-9ecc-11ea-9fc1-0711c3269a45.png"> |  <img width="372" alt="Screen Shot 2020-05-25 at 9 13 47 PM" src="https://user-images.githubusercontent.com/793774/82859918-f0482b80-9ecc-11ea-8c68-151b79e712d6.png"> |


### To test:
1. Launch the app
2. Tap on the Reader tab
3. Tap on the Discover tab
4. For recent posts you should see `{Val}m` or `{Val}h` depending on how new/older the post is
5. Continue scrolling down to the older posts
6. For posts made after a day you should see `{Val}d` 
7. Continue scrolling down to the older posts
8. For posts made more than a week ago you should see the month and day: `May 18`
9. Continue scrolling down to the older posts
10. For posts made in Dec 2019 you should see the full date and year: `Dec 19, 2019` 

PR submission checklist:
